src/lib.rs

```diff
@@ -60,29 +70,59 @@ impl ScuttleButt {
     pub fn self_node_state(&mut self) -> &mut NodeState {
         self.cluster_state_map.node_state_mut(&self.self_node_id)
     }
+
+    /// Retrieve a list of all living nodes.
+    pub fn living_nodes(&self) -> impl Iterator<Item = &str> {
+        self.cluster_state_map.living_nodes()
+    }
+
+    /// Compute digest, always at least adding a heartbeat as update.
+    fn compute_digest(&self) -> Digest {
+        // Ensure for every reply from this node, at least the heartbeat is changed.
+        self.heartbeat += 1;
+        let heartbeat = self.heartbeat;
+        self.self_node_state().set(HEARTBEAT_KEY, heartbeat);
+
+        self.cluster_state_map.compute_digest()
```

Comment:
Added this `compute_digest` method here, which can be used to increment the heartbeat and compute the digest of the rest of the cluster.

That way whenever we compute the digest to check if there's anything that changed, *at the very least* the heartbeat will be a delta of the current node. That should hopefully make this pretty automagic even with potential future changes (assuming nobody accidentally calls the `compute_digest` method directly).

I've considered adding it to the `compute_digest` method in the `ClusterState`, but that doesn't have any notion of its own node and I didn't really like the idea of introducing it. This way the cluster state stays as a generic state for the entire cluster, regardless of where we fall within it.

Reply:
I think that should do the job for the moment.